### ISSUE TITLE
Remove coursier cache from slug

### ISSUE THREAD

**Issue by jkutner** [heroku/heroku-buildpack-scala#131]
[Coursier](https://github.com/coursier/coursier) leaves a large `.sbt_home/.cache/coursier/` directory in the slug. I'm not sure what it's job is, but it can probably be removed from the slug and kept in the build cache.

**Comment by drobert** [comment]
We ran into the same thing. Seems to be a quick fix, I can try to put a PR up tonight, but: https://github.com/heroku/heroku-buildpack-scala/blob/f9e6ab21aca917c536c516ea895f5c74c2e1ec1a/bin/compile#L177-L180 looks for a directory `.coursier` but the directory name, as noted, is `.cache`

**Comment by drobert** [comment]
PR available: https://github.com/heroku/heroku-buildpack-scala/pull/148

**Comment by drobert** [comment]
Incidentally, as a temporary workaround, we've done this in `build.sbt`:

```scala
  .settings(
    // ...
    stageCleanupTask := {
      val localStage = (stage in Universal).value
      val log = streams.value.log
      if (sys.env.getOrElse("POST_STAGE_CLEAN", "true").equals("true")) {
        def cleanup(base: File): PathFinder = {
          // build artifacts typically within a "target" dir, but even
          // the root's aren't needed
          base ** "target" * ("scala-2.12" || "streams") +++
          // any 'target' directory that isn't the root project's
          ((base ** "target") --- (base / "target")) +++
          // ivy2 or coursier lib caches
          base ** (".ivy2" || ".cache")
        }
        sbt.IO.delete(cleanup(baseDirectory.value).get)
      }
      localStage
    },
    stage := stageCleanupTask.value,
    // ...

```

**Comment by Malax** [comment]
This seems to be fixed since 18da9226, closing.
